Source/UI/BillingForm/ViewModel/DefaultBillingFormViewModel.swift

```diff
@@ -173,8 +173,12 @@ final class DefaultBillingFormViewModel: BillingFormViewModel {
             textValueOfCellType[type.index] = nil
         }
 
-        let areAllFieldsAreFulfilled = textValueOfCellType.values.count == style.cells.count && !hasErrorValue
-        editDelegate?.didFinishEditingBillingForm(successfully: areAllFieldsAreFulfilled)
+        let areAllFieldFulfilled = style.cells.first(where: {
+          guard let style = $0.style, style.isMandatory else { return false }
+          return textValueOfCellType[$0.index]?.isEmpty == true
+        }) == nil
+
+        editDelegate?.didFinishEditingBillingForm(successfully: areAllFieldFulfilled && !hasErrorValue)
```

Comment:
```suggestion
        guard !hasErrorValue else {
            editDelegate?.didFinishEditingBillingForm(successfully: false)
            return
        }
        
        let areAllFieldFulfilled = style.cells.first(where: {
          guard let style = $0.style, style.isMandatory else { return false }
          return textValueOfCellType[$0.index]?.isEmpty == true
        }) == nil

        editDelegate?.didFinishEditingBillingForm(successfully: areAllFieldFulfilled)
```

We can avoid iterating through the form altogether if an error value is found early, saving work time.

Reply:
Added in a new [PR](https://github.com/checkout/frames-ios/pull/267)